### DISCUSSION TITLE
Backup Global and per/storage Growing Trends

### DISCUSSION THREAD

**Discussion by heitorfaria** [waa/bacula#10] [Ideas]
Hello Bill,

Please consider including something as:

*query
Automatically selected Catalog: BaculaCatalog
Using Catalog "BaculaCatalog"
Available queries:
     1: Jobs Report 1 day
     2: Total backed up agents
     3: Total backed up data
     4: Last x days Jobs OK
     5: Last x days Jobs with Errors
     6: Last x days Jobs with Fatal Errors
     7: Average Job Duration
     8: Average Transfer Rate (B/s)
     9: Backup Growth Trends
Choose a query (1-9): 9
+-----------------+-----------+
| interval        | gigabytes |
+-----------------+-----------+
| last one day    | 0.00      |
| last one week   | 15.86     |
| last one month  | 142.47    |
| last six months | 164.50    |
| last one year   | 164.50    |
+-----------------+-----------+

SQL:

:Backup Growth Trends
WITH media_trends AS (
  SELECT
    'last one day' AS interval,
    SUM(volbytes) / (1024^3)::NUMERIC AS gigabytes
  FROM media
  WHERE lastwritten >= NOW() - INTERVAL '1 day'

  UNION ALL

  SELECT
    'last one week' AS interval,
    SUM(volbytes) / (1024^3)::NUMERIC AS gigabytes
  FROM media
  WHERE lastwritten >= NOW() - INTERVAL '1 week'

  UNION ALL

  SELECT
    'last six months' AS interval,
    SUM(volbytes) / (1024^3)::NUMERIC AS gigabytes
  FROM media
  WHERE lastwritten >= NOW() - INTERVAL '6 months'

  UNION ALL

  SELECT
    'last one month' AS interval,
    SUM(volbytes) / (1024^3)::NUMERIC AS gigabytes
  FROM media
  WHERE lastwritten >= NOW() - INTERVAL '1 month'

  UNION ALL

  SELECT
    'last one year' AS interval,
    SUM(volbytes) / (1024^3)::NUMERIC AS gigabytes
  FROM media
  WHERE lastwritten >= NOW() - INTERVAL '1 year'
)
SELECT
  interval,
  ROUND(SUM(gigabytes), 2) AS gigabytes
FROM media_trends
GROUP BY interval
ORDER BY
  CASE
    WHEN interval = 'last one day' THEN 1
    WHEN interval = 'last one week' THEN 2
    WHEN interval = 'last one month' THEN 3
    WHEN interval = 'last six months' THEN 4
    WHEN interval = 'last one year' THEN 5
  END;

###########################
Rgds.

**Comment by waa** [comment]
Hello Heitor,

Thank you for this feature request.

I am not fully clear on what is needed here. 

Are you just asking in this discussion for another, separate, small table next to the currently available "Summary" and "Success Rates" tables to show "Backup Growth Rates" like:
```
Backup Growth Rates
 last one day       0.00 GB
 last one week     15.86 GB
 last one month   142.47 GB
 last six months  164.50 GB
 last one year    164.50 GB
```
OR, are you asking for some output, in some format to show each of the items listed in 1-8 too?  Some of those items in 1-8 are already displayed in the report in either the Jobs Table header, or in the "Summary" or "Success Rates" tables. 

Please let me know what you think.

I am looking at the other two discussions you opened too, but I think they are all closely related, so if there is a way to merge these three into one I will do so.

Additionally, I think I will convert this discussion into an actual issue so it can be tracked in Github as such.

**Comment by heitorfaria** [comment]
Hello Bill,
I think we need what is described here:
https://www.veritas.com/content/support/en_US/doc/67898349-133416774-0/v34978207-133416774
https://www.veritas.com/content/support/en_US/doc/67898349-133416774-0/v34978204-133416774
Rgds.
